homeassistant/components/overkiz/entity.py

```diff
@@ -36,7 +38,7 @@ def __init__(
     @property
     def device(self) -> Device:
         """Return Overkiz device linked to this entity."""
-        return self.coordinator.data[self.device_url]
+        return cast(Device, self.coordinator.data[self.device_url])
```

Comment:
I tried to get rid of this one by ading a 'hint' (?) to the DataUpdateCoordinator definition, since I saw this notation in another PR.

```class OverkizDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Device]]):```

This removes the need of this cast (and the other Device cast in executor), however this will add some other errors. Is it ok to keep this cast, or is this an ugly workaround and should it be better to see if we can solve it with the notation above?

```
homeassistant/components/overkiz/coordinator.py:40: error: "dict" is not subscriptable  [misc]
homeassistant/components/overkiz/executor.py:29: error: Redundant cast to "Device"  [redundant-cast]
homeassistant/components/overkiz/__init__.py:112: error: No overload variant of "get" of "Mapping" matches argument type "Optional[str]"  [call-overload]
homeassistant/components/overkiz/__init__.py:112: note: Possible overload variants:
homeassistant/components/overkiz/__init__.py:112: note:     def get(self, key: Union[UIClass, UIWidget]) -> Optional[Platform]
homeassistant/components/overkiz/__init__.py:112: note:     def [_T] get(self, Union[UIClass, UIWidget], Union[Platform, _T]) -> Union[Platform, _T]
homeassistant/components/overkiz/__init__.py:114: error: No overload variant of "get" of "Mapping" matches argument type "Optional[str]"  [call-overload]
homeassistant/components/overkiz/__init__.py:114: note: Possible overload variants:
homeassistant/components/overkiz/__init__.py:114: note:     def get(self, key: Union[UIClass, UIWidget]) -> Optional[Platform]
homeassistant/components/overkiz/__init__.py:114: note:     def [_T] get(self, Union[UIClass, UIWidget], Union[Platform, _T]) -> Union[Platform, _T]
homeassistant/components/overkiz/entity.py:41: error: Redundant cast to "Device"  [redundant-cast]
Found 5 errors in 4 files (checked 13 source files)
```

Reply:
I think it would be great to be able to remove the cast.
For the first error `error: "dict" is not subscriptable  [misc]` make sure that you have `from __future__ import annotations` at the top of the file.
The redundant cast will be easy to solve.

Reply:
I already have `from __future__ import annotations` in coordinator.py. Do you know the name of the `DataUpdateCoordinator[(TYPE)]` notation? I have never seen this before, thus it is hard to research it.

Reply:
Apparently `Dict` from typing is required in this context. That works, down to 2 errors.

```
homeassistant/components/overkiz/__init__.py:112: error: No overload variant of "get" of "Mapping" matches argument type "Optional[str]"  [call-overload]
homeassistant/components/overkiz/__init__.py:112: note: Possible overload variants:
homeassistant/components/overkiz/__init__.py:112: note:     def get(self, key: Union[UIClass, UIWidget]) -> Optional[Platform]
homeassistant/components/overkiz/__init__.py:112: note:     def [_T] get(self, Union[UIClass, UIWidget], Union[Platform, _T]) -> Union[Platform, _T]
homeassistant/components/overkiz/__init__.py:114: error: No overload variant of "get" of "Mapping" matches argument type "Optional[str]"  [call-overload]
homeassistant/components/overkiz/__init__.py:114: note: Possible overload variants:
homeassistant/components/overkiz/__init__.py:114: note:     def get(self, key: Union[UIClass, UIWidget]) -> Optional[Platform]
homeassistant/components/overkiz/__init__.py:114: note:     def [_T] get(self, Union[UIClass, UIWidget], Union[Platform, _T]) -> Union[Platform, _T]
Found 2 errors in 1 file (checked 13 source files)
```

Reply:
That's because inside overkiz you are defining widget and ui_class as string.
```python
@define(init=False, kw_only=True)
class Device:
    widget: str | None = None
    ui_class: str | None = None
```

Should they be defined as UIClass and UIWidget instead?

Reply:
Oh damn, good catch! I hoped a bit that string enums would be compatible. I will check if we can change this in pyoverkiz. My only concern is that it will break if there are new widget/ui_classes introduced (which is outside my control).

However, they are now typed as optional in pyoverkiz and that is not correct.